### PR TITLE
AV-1067 + AV-1062 (SILSIM + Data Logging)

### PR DESCRIPTION
**Pull request by redindelible**
SILSIM depends on the data logging branch's architecture changes, and data logging depends on SILSIM for testing the parser script.

### CODE REVIEW

File: LogParser/cpp_parser.py [hunk 1]
```diff
@@ -0,0 +1,809 @@
+"""
```
Comment (NPIPHI) [inline LogParser/cpp_parser.py:1]: 🤔

**Review by NPIPHI — APPROVED** [review]
LGTM

File: MIDAS/src/data_logging.cpp
Comment (Aadityavoru) [inline MIDAS/src/data_logging.cpp:1]: did not add a gyroscope logger

**Review by Aadityavoru — DISMISSED** [review]
Make changes to add gyroscope and GPS stuff

**Review by aidancostello — APPROVED** [review]

**Pull request merged**